### PR TITLE
make_release.py: Use Branches for Latest 

### PR DESCRIPTION
**Pull request by iguessthislldo**
Part of Fix for #940

Assume `Latest_Major`, `Latest_Minor`, and `Latest_Micro`/`Latest_Beta` and the MPC equivalents are branches instead of tags.

The other part of the fix is actually changing the tags into branches.

Also updated the script to work with Python 3 (it still should work with Python 2.7 though), and tried to do some future proofing, mostly by wrapping the argument parser, because `optparse` is deprecated.

Also did some clean up. Simplifying some code, and using more normal Python patterns.

Use of non read-only global variables should be eliminated, because it's unusual to use them in Python, at least nowadays. I started to do this, but back tracked, because it was taking longer to do so than I would've liked to spend on it.

### CODE REVIEW

**Comment by iguessthislldo** [comment]
On the Codacy and Code Factor errors:
The only thing they're complaining about that I wrote is assigning to `input` which is necessary for working with Python 2 and 3.
The thing about "len(SEQUENCE)" in Codacy is strange because it's the same thing, the only benefit of removing `len` is that it is shorter and maybe you can set the variable to `None` and it could handle that. Everything else has at least some degree of merit generally, but isn't that important in the context of a script like this.

File: ACE/bin/make_release.py [hunk 1]
```diff
@@ -584,30 +608,14 @@ def push ():
...
-                push_latest_tag ("ACE_TAO", "Beta", tagname)
-                push_latest_tag ("ACE_TAO", "Micro", tagname)
-                push_latest_tag ("MPC", "ACETAO_Minor", tagname)
-                push_latest_tag ("MPC", "ACETAO_Micro", tagname)
-            elif opts.release_type == "micro":
-                push_latest_tag ("ACE_TAO", "Beta", tagname)
-                push_latest_tag ("ACE_TAO", "Micro", tagname)
-                push_latest_tag ("MPC", "ACETAO_Micro", tagname)
+            # Push release branches
+            latest_branch_helper (push_latest_branch, opts.release_type)
```
Comment (jwillemsen) [inline ACE/bin/make_release.py:612]: At the moment we release a major also the minor/micro branches needs to be updated, similar for a new minor, that is also a new micro

Comment (iguessthislldo) [reply ACE/bin/make_release.py:612]: Yes, that's what the [`latest_branch_helper` function](https://github.com/DOCGroup/ACE_TAO/pull/1182/files#diff-b13c989b2269a0d6d0f1bc09744a6f70R550) is for:

```python
>>> def latest_branch_helper (fn, release_type):
...     release_types = ("major", "minor", "micro")
...     do = release_types[release_types.index(release_type):]
...     if "micro" in do:
...         fn ("ACE_TAO", "Beta")
...         fn ("ACE_TAO", "Micro")
...         fn ("MPC", "ACETAO_Micro")
...     if "minor" in do:
...         fn ("ACE_TAO", "Minor")
...         fn ("MPC", "ACETAO_Minor")
...     if "major" in do:
...         fn ("ACE_TAO", "Major")
...         fn ("MPC", "ACETAO_Major")
... 
>>> latest_branch_helper(print, "micro")
ACE_TAO Beta
ACE_TAO Micro
MPC ACETAO_Micro
>>> latest_branch_helper(print, "minor")
ACE_TAO Beta
ACE_TAO Micro
MPC ACETAO_Micro
ACE_TAO Minor
MPC ACETAO_Minor
>>> latest_branch_helper(print, "major")
ACE_TAO Beta
ACE_TAO Micro
MPC ACETAO_Micro
ACE_TAO Minor
MPC ACETAO_Minor
ACE_TAO Major
MPC ACETAO_Major

```

**Comment by iguessthislldo** [comment]
@jwillemsen these are the two kinds of commands I've been using to test the script:

```
python3 ../doc/ACE_TAO/ACE/bin/make_release.py \
  --major \
  --root git@github.com:iguessthislldo-test-org/ACE_TAO.git \
  --mpc_root git@github.com:iguessthislldo-test-org/MPC.git \
  --verbose \
  --update \
  --tag \
  --push

python3 ../doc/ACE_TAO/ACE/bin/make_release.py \
  --verbose \
  --kit
```

I have tested switching `--major` to `--minor` and `--micro`. They seem to be working properly as far I can tell. Is there any other combination of arguments you think I should test out?

**Comment by jwillemsen** [comment]
Commands I use are all listed in https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/docs/bczar/bczar.html.

**Comment by jwillemsen** [comment]
Is python3 adviced? If so, please update bczar.html, from there I copy the commands

**Comment by iguessthislldo** [comment]
> Commands I use are all listed in https://github.com/DOCGroup/ACE_TAO/blob/master/ACE/docs/bczar/bczar.html.

I wasn't aware of that file. The arguments I figured out from trial and error and reading the script are the same as what bczar.html says to run though.

> Is python3 adviced? If so, please update bczar.html, from there I copy the commands

It's not required. I made it compatible with Python 3+, but also kept compatibility with at least 2.7, which I also tested with. I can update bczar.html to say that just to be safe.

**Comment by jwillemsen** [comment]
Please update the script and bczar.html to use and mention python3, that should work on my opensuse box, no need for backward compatibility for this script which is only used forca release

**Comment by iguessthislldo** [comment]
Okay, then do you want me to rip out the compatibility wrappers/shims for Python 2? I can also change the shebang in the script, so the commands in the bczar stay the same, but I would still mention it requires Python 3.

**Comment by jwillemsen** [comment]
Personally I would like to keep it simple, so drop python2 support, what do you think @mitza

**Comment by mitza-oci** [comment]
> 
> 
> Personally I would like to keep it simple, so drop python2 support, what do you think @mitza-oci

I'm OK with dropping support for Python 2

**Comment by iguessthislldo** [comment]
Alright, it's not much code to support it, but I'm fine with dropping it.

**Comment by jwillemsen** [comment]
I would propose to release 6.5.11 soon after you have implemented this so that we have a real release tested with all changes fresh in memory

**Comment by iguessthislldo** [comment]
Alright, then I think it's ready to be review and merged. After that I will create the branches here and in MPC and move on to doing the same to DAnCE and CIAO.

**Pull request merged**